contribs/analysis/src/main/java/org/matsim/contrib/analysis/vsp/traveltimedistance/HereMapsRouteValidator.java

```diff
@@ -101,10 +93,10 @@ public HereMapsRouteValidator(String outputFolder, String appId, String appCode,
 		DecimalFormat df = (DecimalFormat) NumberFormat.getNumberInstance(locale);
 		df.applyPattern(pattern);
 
-		String urlString = "http://route.cit.api.here.com/routing/7.2/calculateroute.json?app_id=" + appId
-				+ "&app_code=" + appCode + "&waypoint0=geo!" + df.format(from.getY()) + "," + df.format(from.getX())
-				+ "&waypoint1=geo!" + df.format(to.getY()) + "," + df.format(to.getX()) + "&departure=" + date + "T"
-				+ Time.writeTime(trip.getDepartureTime()) + "&mode=fastest;car;traffic:enabled";
+		String urlString = "https://router.hereapi.com/v8/routes?" + "&apiKey=" + apiAcessKey + "&transportmode=car&origin="
```

Comment:
I suggest using https://developer.here.com/documentation/routing/dev_guide/topics/resource-calculate-matrix.html for this type of analysis.

For a set of nodes (zone centroids) and selected departure times, we could compute:
 - skim matrices based on matsim runs (e.g. using matsim-sbb-extensions)
 - matrices with HERE Matrix API

HERE Matrix API is limited to 15x100 matrices per call. Let's say we have 200 zones, then we need only 28 calls to fill up the whole matrix for a given departure time. It should take only a few seconds when making all 28 calls in parallel.

Reply:
@luchengqi7 I can help you with it if we decide to go that way (and if we want to invest time on that).

Reply:
Thank you very much for the feedback!

Yes. I think it will be a good idea to drop the change in the ParallelEventManager. It is just a tempeory fix. I will revert that commit. 

As for the HERE API, this is in another project. This validator compares the travel time and distance of a trip in our simulation to the same trip based on HERE database. The validator is ran seperately without MATSim iteration process. It intake the event file and extract trips from the event file. Then the trip will be compared to the data acquired from the HERE API. I think we will need the exact departure point for that validation. So I think it is a little bit different from the zonal based things. But that can be a good idea when deciding the cost for traveling between zones at different time of the day. Perhaps that can be a future implementation in the rebalancing/matching strategy.

By the way, do we also use that validator for other purposes (e.g. during or between MATSim iteration)? Because it seems that the original run code does not initialize the CarTripExtractor (an implemntation of event handler).

Reply:
Uh oh, I think I failed to drop that comit as the comit also covers other files. I will stash the commit at the end.

Reply:
Thanks for your explanations.

Anyway, if we would like to validate matsim TTs against HERE API in the future, I am happy to join.